flytekit/core/node.py

```diff
@@ -194,6 +194,8 @@ def with_overrides(
 
         if name is not None:
             self._metadata._name = name
+            if self.run_entity and hasattr(self.run_entity, "metadata"):
```

Comment:
We should not override the run_entity.

Reply:
Suppose you have two tasks in a workflow, like below. The cache version of the second task will be overridden as well.
```
@workflow
def wf():
  t1().with_override(cache_version="v2")
  t1()
```

Reply:
You can pass the node info to `local_execute`, and use the cache version in the node info to retrieve the data from the cache.
https://github.com/flyteorg/flytekit/blob/935be5c28dccbf860f7188357edc3bba9c7c70f1/flytekit/core/base_task.py#L306

Reply:
@pingsutw Could I know how to do this?

"You can pass the node info to local_execute`

Reply:
I think the problem is: 

In `node.py`, `self._metadata._cacheable` won't be affecting the actual Task entity.

That's why we want to modify the Task's entity's metadata.

But I acknowledge the issue that the two tasks example woulld be problematic. I am not sure how `local_config` can help with it.